test/myxql/client_test.exs

```diff
@@ -447,4 +464,57 @@ defmodule MyXQL.ClientTest do
 
     %{pid: pid, port: port}
   end
+
+
+  defp start_cleartext_fake_server() do
+    start_fake_server(fn %{accept_socket: sock} ->
+      initial_handshake =
+        <<74, 0, 0, 0, 10, 56, 46, 48, 46, 51, 53, 0, 127, 24, 4, 0, 93, 42, 61, 27, 60,
+          38, 85, 12, 0, 255, 255, 255, 2, 0, 255, 223, 21, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+          0, 39, 48, 10, 117, 54, 65, 74, 37, 125, 121, 93, 6, 0, 109, 121, 115, 113,
+          108, 95, 110, 97, 116, 105, 118, 101, 95, 112, 97, 115, 115, 119, 111, 114,
+          100, 0>>
```

Comment:
this is tough to understand and maintain. Could you break it down to along the lines of:

```elixir
[
  <<74, 0, 0, 0>>,
  [
    # protocol version, always 0x10
    10,
    # server version
    <<"8.0.35", 0>>,
    # ...
  ]
]
```

I'm really curious to see where exactly the scramble was supposed to be and is empty.

Reply:
Sure, but just FYI this part doesn't matter. It's just priming the connection with a scramble, that we are going to make (knowingly) an unsuccesful authentication with. When that fails we continue with the cleartext password. This came from painfully figuring out why I couldn't just send the cleartext password with the `mysql_clear_password` attached to it. It can only happen with the auth switch.